src/ECharts/Image.purs

```diff
@@ -0,0 +1,49 @@
+module ECharts.Image (
+  ImgType(..),
+  getDataURL,
+  getImage
+  ) where 
+
+import DOM
+import Control.Monad.Eff
+import Data.Function
+import Data.Argonaut.Core
+import Data.Argonaut.Encode
+
+import ECharts.Chart 
+import ECharts.Effects
+
+
+data ImgType = PNG | JPEG
+
+instance imgTypeShow :: Show ImgType where
```

Comment:
Ditto, Show should only be used to render as Purescript code, functions should be used for other use cases. I won't comment on the rest of these.